core/observables/observable.py

```diff
@@ -118,7 +118,7 @@ def from_string(string):
         return results
 
     @classmethod
-    def add_text(cls, text, tags=[]):
+    def add_text(cls, text, tags=[], type=None):
```

Comment:
`type` exists in Python. Maybe you could use `typename` (also, it shows clearly that you expect a name, not the Python type)?

Reply:
I'd actually change this variable name to `force_type` to indicate that we're overriding the guessing functionality.

Reply:
I changed the variable name